client/state/preferences/reducer.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import has from 'lodash/has';
+import get from 'lodash/get';
+import forOwn from 'lodash/forOwn';
+/**
+ * Internal dependencies
+ */
+import {
+	PREFERENCES_SET,
+	PREFERENCES_RECEIVE,
+	PREFERENCES_REMOVE,
+	PREFERENCES_FETCH
+} from 'state/action-types';
+import { USER_SETTING_KEY } from './actions';
+
+function values( state = {}, action ) {
+	if ( action.type === PREFERENCES_RECEIVE && has( action, [ 'data', USER_SETTING_KEY ] ) ) {
+		state = Object.assign( {}, state, get( action, [ 'data', USER_SETTING_KEY ] ) );
+		forOwn( state, ( value, key ) => {
+			if ( null === value ) {
+				delete state[ key ];
+			}
+		} );
+	} else if ( action.type === PREFERENCES_SET && action.key && action.value ) {
+		state = Object.assign( {}, state );
+		state[ action.key ] = action.value;
+	} else if ( action.type === PREFERENCES_REMOVE && action.key ) {
+		state = Object.assign( {}, state );
+		state[ action.key ] = null;
+	}
+	return state;
+}
+
+function isFetching( state = false, action ) {
+	switch ( action.type ) {
+		case PREFERENCES_RECEIVE:
```

Comment:
The problem with using `PREFERENCES_RECEIVE` as an indicator that fetching is completed is that it opens up the possibility that a `save` occurs and finishes before the fetch completes, causing the `isFetching` state to wrongly change to `false`. I'd recommend having separate actions for indicating the completion of the fetch specifically.

Example: https://github.com/Automattic/wp-calypso/blob/1f10439/client/state/post-types/actions.js#L43-L47